### DISCUSSION TITLE
Run as a system service

### DISCUSSION THREAD

**Discussion by enigmathion** [owncast/owncast#873] [General Owncast discussion]
Hi, I'm trying to get it to run as a service and I can't do it, maybe because I'm putting the data wrong! If the owncast folder is in the root of the server and the owncast executable is inside that folder, how should I complete:
WorkingDirectory = [path_to_owncast_root_directory]
ExecStart = [path_to_owncast_executable]
Thanks a lot!

**Comment by jeyemwey** [comment]
Have you tried

WorkingDirectory=/owncast 
ExecStart=/owncast/owncast

**Comment by enigmathion** [comment]
Yes, and not found! Any ideas?

**Reply by gabek** [reply]
What's the actual error message you're seeing that you're trying to resolve?

Also, I don't know if it's required, but you might want to specify a username and user group that matches a user who has access to that location.

```
User=owncast
Group=owncast
```

**Comment by enigmathion** [comment]
Although the user is the same root? The installation was done manually!

**Comment by enigmathion** [comment]
I simply run:
owncast cd
./owncast
but the service stops only at hours!

**Reply by gabek** [reply]
I think you should read the documentation of systemd and learn the basics of how it works. You're trying to setup a systemd service, but trying to run Owncast outside of systemd.

If configured properly systemd will start owncast when you start your machine as a background service. You wouldn't want to run owncast manually.

**Comment by enigmathion** [comment]
I fully understand what you are telling me, the issue is that for some
reason that I am trying to solve, it does not start! That's why I end up
activating manually so that it works perfectly! Is there an installation
script that executes all the necessary configurations?

**Reply by gabek** [reply]
I'm no systemd expert.  It has nothing to do with Owncast, so I don't have the answers of how you should set it up specifically on your machine.  I simply suggest you look at their documentation and not the Owncast documentation.  While you an run Owncast under systemd, the specifics of how systemd is running on your machine isn't something that we can help with.

https://www.golinuxcloud.com/beginners-guide-systemd-tutorial-linux/
